WORKSPACE

```diff
@@ -97,14 +106,31 @@ load("@io_bazel_rules_sass//:defs.bzl", "sass_repositories")
 
 sass_repositories()
 
+# Always bump the requirements.txt protobuf dep to be >= the version here.
+# Keep this version to be in sync with TensorFlow:
+# https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace2.bzl#L446
+# For other projects (e.g. Keras) depending on tb-nightly, generating protobuf code at
+# a more recent version than the protobuf runtime supplied by TF's bazel build tooling
+# might lead to test failures.
+http_archive(
+    name = "com_google_protobuf",
+    sha256 = "1fbf1c2962af287607232b2eddeaec9b4f4a7a6f5934e1a9276e9af76952f7e0",
+    strip_prefix = "protobuf-3.9.2",
+    urls = [
+        "http://mirror.tensorflow.org/github.com/protocolbuffers/protobuf/archive/v3.9.2.tar.gz",
+        "https://github.com/protocolbuffers/protobuf/archive/v3.9.2.tar.gz",  # 2019-09-23
+    ],
+)
+
 # gRPC.
 http_archive(
     name = "com_github_grpc_grpc",
-    sha256 = "b2f2620c762427bfeeef96a68c1924319f384e877bc0e084487601e4cc6e434c",
-    strip_prefix = "grpc-1.42.0",
+    sha256 = "b956598d8cbe168b5ee717b5dafa56563eb5201a947856a6688bbeac9cac4e1f",
+    strip_prefix = "grpc-b54a5b338637f92bfcf4b0bc05e0f57a5fd8fadd",
     urls = [
-        "http://mirror.tensorflow.org/github.com/grpc/grpc/archive/v1.42.0.tar.gz",
-        "https://github.com/grpc/grpc/archive/v1.42.0.tar.gz",  # 2021-11-17
+        # Same as TF: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace2.bzl#L492
```

Comment:
Maybe worth a note that we can't upgrade GRPC past 1.30.0 without also bumping protobuf?

(It's not clear that this commit that TF uses is any particular version, but it looks like it's roughly 1.27.0)

Reply:
Done.